### PR TITLE
Upgraded PHPUnit to version 9.3.0, PHPStan to version 0.12.0 and bumped the minimum version to PHP 7.3

### PR DESCRIPTION
**Pull request by jensdenies**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### Description

Upgraded PHPUnit to version 9.3.0, PHPStan to version 0.12.0 and bumped the minimum version to PHP 7.3

### CODE REVIEW

**Comment by jensdenies** [comment]
@lcobucci I've created this PR just for PHPUnit.

File: composer.json [hunk 1]
```diff
@@ -51,9 +51,10 @@
         "mezzio/mezzio-fastroute": "^3.0",
         "mezzio/mezzio-laminasrouter": "^3.0",
         "mockery/mockery": "^1.0",
-        "phpstan/phpstan": "^0.9.2",
-        "phpstan/phpstan-strict-rules": "^0.9",
-        "phpunit/phpunit": "^7.0.1"
+        "phpstan/phpstan": "^0.12.0",
+        "phpstan/phpstan-strict-rules": "^0.12",
```
Comment (jensdenies) [inline composer.json:55]: Those were necessary because the previous version conflicted with PHPunit 9.3.

**Comment by lcobucci** [comment]
@jensdenies ouch, I just saw this now (after pushing #59).

I took a quite different approach by removing prophecy and all attribute-based assertions. Do you agree with using my solution instead?

**Comment by jensdenies** [comment]
@lcobucci Haha, no worries! Yes, your solution is better because instead of polyfilling, it actually fixes it. 👍🏼

**Pull request closed without merging**